### DISCUSSION TITLE
reqtest: Add RecorderMode

### DISCUSSION THREAD

**Discussion by earthboundkid** [earthboundkid/requests#123] [Ideas]
When writing tests you want to record responses, but then once you get done writing tests, you want it to be an error to issue a request without a corresponding entry. You can do this by just switching from reqtest.Caching to reqtest.Replay, but it would be more convenient if there were a single function that just took one of three mode arguments.

**Comment by acehinnnqru** [comment]
Do you have any code sample?

**Reply by earthboundkid** [reply]
See #124.
